src/wp-includes/comment-template.php

```diff
@@ -1889,16 +1889,32 @@ function post_reply_link( $args = array(), $post = null ) {
  *
  * @since 2.7.0
  *
- * @param string $text Optional. Text to display for cancel reply link. If empty,
- *                     defaults to 'Click here to cancel reply'. Default empty.
+ * @param string $text    Optional. Text to display for cancel reply link. If empty,
+ *                        defaults to 'Click here to cancel reply'. Default empty.
+ * @param int    $post_id Optional. The post ID the comment thread is being
+ *                        displayed for. Default: current global post ID.
  * @return string
  */
-function get_cancel_comment_reply_link( $text = '' ) {
+function get_cancel_comment_reply_link( $text = '', $post_id = 0 ) {
 	if ( empty( $text ) ) {
 		$text = __( 'Click here to cancel reply.' );
 	}
 
-	$style = isset( $_GET['replytocom'] ) ? '' : ' style="display:none;"';
+	if ( empty( $post_id ) ) {
+		$post_id = get_the_ID();
+	}
+
+	$reply_to_id = isset( $_GET['replytocom'] ) ? (int) $_GET['replytocom'] : 0;
+
+	if ( 0 !== $reply_to_id ) {
+		$parent_comment = get_comment( $reply_to_id );
+
+		if ( ! $parent_comment || 0 === (int) $parent_comment->comment_approved || $post_id !== (int) $parent_comment->comment_post_ID ) {
+			$reply_to_id = 0;
+		}
+	}
```

Comment:
This code is duplicated more than once in this PR. It has a specific purpose to get the reply to ID and validate it. That business logic is a separate task from these functions. I'd suggest abstracting this business logic to a private function such as:

```php
**
 * Gets the comment's reply to ID from the $_GET['replytocom'].
 *
 * @since 5.9.0
 * @access private
 *
 * @param int $post_id Post ID.
 * @return int Comment's reply to ID.
 */
function _get_comment_reply_id( $post_id ) {
	if ( ! isset( $_GET['replytocom'] ) || ! is_numeric( $_GET['replytocom'] ) ) {
		return 0;
	}

	$reply_to_id = (int) $_GET['replytocom'];

	/*
	 * Validate the comment.
	 * Bail out if it does not exist, is not approved, or its
	 * `comment_post_ID` does not match the given post ID.
	 */
	$comment = get_comment( $reply_to_id );
	if ( ! $comment instanceof WP_Comment ||
		 0 === (int) $comment->comment_approved ||
		 $post_id !== (int) $comment->comment_post_ID ) {
		return 0;
	}

	return $reply_to_id;
}
```

Reply:
Good call! I've updated my local `comment-template.php` with this and I'll get it pushed when we've resolved the other conversations.